### ISSUE TITLE
Incorrect nav applied - Super Admin group now has restricted nav

### ISSUE THREAD

**Issue by peteeveleigh** [engram-design/CPNav#83]
It appears as if the most recent nav layout is being applied to all user groups regardless of plugin settings.

While logged in as a Super Admin, I created a very minimal nav layout for a specific user group.
The nav changed while editing to show the minimal layout.
When completed I left the plugin settings but the minimal nav was still applied.
Logging out and back in again (as Super Admin) still only shows minimal nav.
Clearing all site data and logging back in does not fix the problem.

If I log in as a user in another member group (that should have a different customised nav layout) the minimal layout is being applied.

This appears to be related to turning off the "standard" nav items via the lightswitches in the plugin settings. The state of the lightswitch seems to apply to *all* layouts instead of only the one being edited.

- Plugin version: 3.0.13.1
- Craft version: 3.15.1
- Multi-site: No

**Comment by peteeveleigh** [comment]
I've managed to restore my original nav by deleting the new minimal nav from the craft_cp_nav_layout table

**Comment by engram-design** [comment]
I can't seem to immediately replicate this. Are you able to provide some information about your layout and user group assignment setup? I don't suppose the Super Admin account was assigned to the same user group as the user you wanted to set the nav items for?

If you're able to replicate again, it'd be worth looking at your `cpnav_navigation` table to see if the correct layout IDs are being applied to your nav items.

There's also a quick "Reset Navigation" tab  which can help regenerate the original navs if you get stuck.

**Comment by peteeveleigh** [comment]
Hi Josh,

I tried it a few times and got the same issue each time. I don’t think the Super Admin account was assigned to other groups too but I will check that again.

It did seem to be related to turning off the regular nav options using the lightswitches. The settings for those seemed to be applied to all navigation layouts regardless of on which layout they were set.

I’ll do some more testing with it and let you know what I can figure out.

Pete

**Comment by mikeymeister** [comment]
I experienced a similar, if not the same, issue to this where the superadmin was locked into another layout. It was due to the default layout not having an id of 1. I'm not sure how I managed to get into that situation but I had the issue following some updates and failed project-config syncs. In `LayoutsService::getLayoutForCurrentUser()` the function returns the layout with an id of 1 if the user isn't in any groups that apply to layout permissions.

Replacing `return $this->getLayoutById(1)` with:
`return ArrayHelper::firstWhere($this->getAllLayouts(), 'isDefault','1');`
seemed to fix this issue in my case.
